language/tools/move-cli/src/sandbox/utils/on_disk_state_view.rs

```diff
@@ -72,6 +80,56 @@ impl OnDiskStateView {
         Ok(path.into_os_string().into_string().unwrap())
     }
 
+    pub(crate) fn interface_files_metadata_file(&self) -> PathBuf {
+        // File containing interface file metadata, specifically named address mapping
+        const INTERFACE_FILES_METADATA: &str = "metadata";
+        let mut path = self.build_dir.join(MOVE_COMPILED_INTERFACES_DIR);
+        path = path.join(INTERFACE_FILES_METADATA);
+        path.set_extension("txt");
```

Comment:
oof yeah. `txt` was a placeholder until I decided on BCS for the encoding. It might actually be better to do some sort of JSON encoding if possible so it can be readable. @tzakian, any ideas?

Reply:
I think having something human readable here is a good idea since we don't have space issues/need canonical serialization for this. 

My plan for the packages is this type of compiled metadata would be held in JSON encoding, so am partial to that, but if you prefer a different encoding we can go for that as well (the only other one that I can think of as a good contender might be YAML?).

Reply:
I found `serde_json` which makes my life easy for that. Is there something similar for YAML? just to compare

Reply:
```
{
  "named_address_mapping": {
    "00000000000000000000000000000042": {
      "M": "A",
      "N": "A"
    }
  }
}
```
This is what the file will look like. A lot better than BCS given how small it is :)

Reply:
There is! It is given the very inventive name of [serde_yaml](https://docs.rs/serde_yaml/0.8.17/serde_yaml/) :p

Reply:
ooo yaml is even better as I don't have to do this weird json transformation:
```
---
named_address_mapping:
  ? address: "00000000000000000000000000000042"
    name: M
  : A
  ? address: "00000000000000000000000000000042"
    name: N
  : A
```